engine/src/main/java/com/odysseusinc/arachne/executionengine/service/impl/DescriptorServiceImpl.java

```diff
@@ -97,12 +113,29 @@ public DescriptorBundle getDescriptorBundle(File dir, Long analysisId, String re
                         findRequestedDescriptor(analysisId, available, id)
                 ).orElseGet(() -> {
                     LOGGER.info("For analysis [{}] no descriptor is requested explicitly, fall back to dependency matching among {} present descriptors", analysisId, available.size());
-                    return findMatchingDescriptor(dir, analysisId, available);
+                    return findMatchingDescriptor(dir, analysisId, available, requestedDescriptorId);
                 })
-        ).orElseGet(() -> {
+        ).orElseGet(getDefaultBundle(analysisId, requestedDescriptorId));
+    }
+
+    private Supplier<DescriptorBundle> getDefaultBundle(Long analysisId, String requestedDescriptorId) {
+        return () -> {
             LOGGER.info("For analysis [{}] using default descriptor (no descriptors configured). Requested [{}]", analysisId, requestedDescriptorId);
-            return defaultDescriptorBundle;
-        });
+            DescriptorBundle defaultBundle = new DescriptorBundle(archiveFolder.map(Path::toString).orElse(""), new DefaultDescriptor());
+            return Optional.ofNullable(defaultDescriptorFile)
+                    .map(f -> archiveFolder.map(a -> a.resolve(f))
+                            .filter(Files::isRegularFile)
+                            .map(Path::toFile)
+                            .map(deserializeDescriptor())
+                            .map(this::toBundle)
+                            .orElseGet(() -> {
+                                LOGGER.warn("*** Property runtimeservice.dist.archiveFolder is not set or not exist. Empty default descriptor will used.");
+                                return defaultBundle;
+                            })).orElseGet(() -> {
+                        LOGGER.warn("*** Property runtimeservice.dist.defaultDescriptorFile is not set. Empty default descriptor will used.");
```

Comment:
I still think that reporting a dummy descriptor is fundamentally incorrect. The execution will not work anyway, so why not fail fast. If Execution engine has no working default, this is exactly what needs to be reported to client. And client has to live with that.